docs/manual/interning-checker.tex

```diff
@@ -148,40 +158,95 @@
 The Interning Checker ensures that only interned
 values can be assigned to \code{s2}.
 
-To specify that \emph{all} objects of a given type are interned, annotate the
-class declaration:
+\sectionAndLabel{Interned classes}{interning-interned-classes}
+
+An interned annotation on a class declaration indicates that all objects of a
+type are interned \textit{except for newly created objects}. That means that
+all uses of such types are \<@Interned> by default and the type \<@UnknownInterned
+MyClass> is an invalid type.
+
+An exception is \textit{constructor results}. Constructor results and \<this> within the
+body of the constructor are \<@UnknownInterned> by default, but no ``type.invalid''
+error is issued at the declaration. (It would be issued because
+\<@UnknownInterned InternedClass> is not a legal type, given the
+qualifier \<@Interned> on \<InternedClass>.)  Instead, an ``interned.object.creation''
+error is issued at the invocation of the constructor. The user should inspect
+this location and suppress the warning if the newly created object is interned.
+For example:
 
 \begin{Verbatim}
-  public @Interned class MyInternedClass { ... }
+@Interned class InternedClass {
+  @UnknownInterned InternedClass() {
+  // error, "this" is @UnknownInterned.
+  @Interned InternedClass that = this;
+  }
+
+  @SuppressWarnings("intern") // ONE is interned for some reason.
```

Comment:
This is a bit confusing, too, because it's very vague. I'd give an example of a reason, rather than saying "some reason"